### PR TITLE
Meta: Fix `mke2fs` on MacOS

### PR DESCRIPTION
**Pull request by EWouters**
Use `Meta/.shell_include.sh` to find the `mke2fs` executable.

### CODE REVIEW

**Comment by timschumi** [comment]
We now have infrastructure for finding executables in place in `.shell_include.sh`, maybe we should just add mke2fs there instead of proceeding with handling every case manually.

**Comment by EWouters** [comment]
Oh, I'll check it out. I'm also not sure if `brew` installs `e2fsprogs` in the same location on older MacOS.

File: Meta/build-image-qemu.sh [hunk 1]
```diff
@@ -122,7 +122,7 @@ if [ $USE_EXISTING -ne 1 ]; then
         if [ -x /sbin/mke2fs ]; then
```
Comment (EWouters) [inline Meta/build-image-qemu.sh:122]: Not sure if we need the `if`-statement at all now.

File: Meta/.shell_include.sh [hunk 2]
```diff
@@ -20,9 +20,8 @@ die() {
 find_executable() {
   paths=("/usr/sbin" "/sbin")
 
-  if [ "$(uname -s)" = "Darwin" ]; then
-    paths+=("/usr/local/opt/e2fsprogs/bin" "/usr/local/opt/e2fsprogs/sbin")
-    paths+=("/opt/homebrew/opt/e2fsprogs/bin" "/opt/homebrew/opt/e2fsprogs/sbin")
+  if [ "$(uname -s)" = "Darwin" ] && [ -n "$HOMEBREW_PREFIX" ]; then
+    paths+=("$HOMEBREW_PREFIX/opt/e2fsprogs/bin" "$HOMEBREW_PREFIX/opt/e2fsprogs/sbin")
```
Comment (timschumi) [inline Meta/.shell_include.sh:24]:
```suggestion
  if [ "$(uname -s)" = "Darwin" ] && [ -n "${HOMEBREW_PREFIX}" ]; then
    paths+=("${HOMEBREW_PREFIX}/opt/e2fsprogs/bin" "${HOMEBREW_PREFIX}/opt/e2fsprogs/sbin")
```

If I remember my bash expansions correctly we could also compress this to the following:

```suggestion
  if [ "$(uname -s)" = "Darwin" ] && [ -n "${HOMEBREW_PREFIX}" ]; then
    paths+=("${HOMEBREW_PREFIX}/opt/e2fsprogs/{bin,sbin}")
```

In any case, can we be sure that `HOMEBREW_PREFIX` is always set?

File: Meta/build-image-qemu.sh (hunk 1 above)
Comment (timschumi) [reply Meta/build-image-qemu.sh:122]: Yes, that can be removed. `/sbin` is covered by `find_executable`, and the actual arguments also don't seem to differ.

File: Meta/.shell_include.sh (hunk 2 above)
Comment (EWouters) [reply Meta/.shell_include.sh]: I'm also not sure if `paths+=("${HOMEBREW_PREFIX}/opt/e2fsprogs/{bin,sbin}")` will expand correctly, so I'll omit that for now.

**Comment by EWouters** [comment]
I'll refrain from fixing `"${}"` on lines I don't need to touch...

**Comment by timschumi** [comment]
All good, just wanted to make sure that we don't add any new ones without `${}`. The change itself looks good to me, let's see if CI passes...

**Review by ADKaster — APPROVED** [review]

**Pull request merged**